tools/clang/lib/SPIRV/LiteralTypeVisitor.cpp

```diff
@@ -294,17 +294,9 @@ bool LiteralTypeVisitor::visit(SpirvVectorShuffle *inst) {
   return true;
 }
 
-bool LiteralTypeVisitor::visit(SpirvNonUniformUnaryOp *inst) {
-  // Went through each non-uniform binary operation and made sure the following
-  // does not result in a wrong type deduction.
-  tryToUpdateInstLitType(inst->getArg(), inst->getAstResultType());
-  return true;
-}
-
-bool LiteralTypeVisitor::visit(SpirvNonUniformBinaryOp *inst) {
-  // Went through each non-uniform unary operation and made sure the following
-  // does not result in a wrong type deduction.
-  tryToUpdateInstLitType(inst->getArg1(), inst->getAstResultType());
+bool LiteralTypeVisitor::visit(SpirvGroupNonUniformOp *inst) {
+  for (auto *operand : inst->getOperands())
```

Comment:
In the former `SpirvNonUniformBinaryOp`, only the 2nd arg (`inst->getArg1()`) was used as the argument to `tryToUpdateInstLitType`. Is the idea to use the last argument, rather than iterate through all of them?

Reply:
This is another bit of cargo culting of untested code that I think was either buggy or possibly totally unnecessary. I will try to come up with some HLSL to trigger it and include a test...